### PR TITLE
TST: Parameterize & make tests more performant

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/tests/frame/methods/test_align.py [hunk 1]
```diff
@@ -392,27 +392,57 @@ def test_missing_axis_specification_exception(self):
...
+        "right_slice",
+        [
+            [slice(2, None), slice(6, None)],
+            [slice(0), slice(0)],
+        ],
+    )
+    def test_align_fill_method(
+        self, how, method, axis, fill_axis, float_frame, left_slice, right_slice
+    ):
+        limit = 1
```
Comment (WillAyd) [inline pandas/tests/frame/methods/test_align.py:416]: Are we missing a case where `limit = None` from the original `_check_align_fill` implementation?

**Review by WillAyd — COMMENTED** [review]
nice changes. do you know how much time this saves?

File: pandas/tests/io/test_pickle.py [hunk 2]
```diff
@@ -190,29 +196,27 @@ def python_unpickler(path):
...
-
-                # and the same for file objects (GH 35679)
-                with open(path, mode="wb") as handle:
-                    writer(expected, path)
-                    handle.seek(0)  # shouldn't close file handle
-                with open(path, mode="rb") as handle:
-                    result = pd.read_pickle(handle)
-                    handle.seek(0)  # shouldn't close file handle
-                compare_element(result, expected, typ)
+@pytest.mark.parametrize("typ, expected", flatten(create_pickle_data()))
```
Comment (jbrockmendel) [inline pandas/tests/io/test_pickle.py:199]: how big is the memory footprint of create_pickle_data?  i think doing this outside the test means it sticks around during the whole collection/runtime

File: pandas/tests/frame/methods/test_align.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/frame/methods/test_align.py]: Good catch! Add a `parametrize` over `limit = [None, 1]`

**Review by WillAyd — APPROVED** [review]
lgtm when green

**Comment by mroeschke** [comment]
> do you know how much time this saves?

Locally in total for this PR probably a second or so. Another benefit for the tests that reduce the data size is to also to reduce the peak memory pressure of a test.

File: pandas/tests/io/test_pickle.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/io/test_pickle.py:199]: Doing a recursive `getsizeof` of the dict yields 426965 bytes so it doesn't seem too much. Additionally IIUC before this was being yielded from the `current_pickle_data` fixture which caches & keeps the data around after first use

**Pull request merged**